mkdocs/config/config_options.py

```diff
@@ -343,3 +343,66 @@ def post_validation(self, config, key_name):
             config[key_name] = len(config['pages']) > self.at_lest
         except TypeError:
             config[key_name] = False
+
+
+class Private(BaseConfigOption):
+    """
+    Private Config Option
+
+    A config option only for internal use. Raises an error if set by the user.
+    """
+
+    def run_validation(self, value):
+        raise ValidationError('For internal use only.')
+
+
+class MarkdownExtensions(BaseConfigOption):
+    """
+    Markdown Extensions Config Option
+
+    A list of extensions. If a list item contains extension configs,
+    those are set on the private  setting passed to `configkey`. The
+    `builtins` keyword accepts a list of extensions which cannot be
+    overriden by the user. However, builtins can be duplicated to define
+    config options for them if desired.
+    """
+    def __init__(self, builtins=None, configkey='mdx_configs', **kwargs):
+        super(MarkdownExtensions, self).__init__(**kwargs)
+        self.builtins = builtins or []
+        self.configkey = configkey
+        self.configdata = {}
+
+    def run_validation(self, value):
+        if isinstance(value, dict):
+            # Delete this if block when dict support is removed
```

Comment:
I've been trying to add `# TODO: remove in 1.0` to these. It could happen sooner, potentially, but 1.0 would be the latest this should happen.